config/configurator.go

```diff
@@ -0,0 +1,44 @@
+package config
+
+import "github.com/spf13/viper"
+
+type configurationLoader interface {
+	setConfig(name string, extension string, path string)
+	read() error
+	merge() error
```

Comment:
I'm not sure about `merge` - to me it's something related with the design of `viper`, for other vendor packages(s) there may not be such concept. I would say this whole `setName -> read -> setAnotherName -> merge` thing is viper specific, and should be handled in a different layer.

Reply:
I think we can put `setConfig`, `read`, and `merge` together to `loadConfig(filename string)`, because from the high level view, what `config` needs to do is to convert config files into native struct. Low level provider should take a filename, and parse and persist the actual config key/value pairs (to allow `getBool` and `getString` later), and should allow high level caller to do:

- load -> load another -> load another ...
- possibly provide a "clear" function, which is not a business requirement now

i.e. `merge` should be the default behavior as that's the usual use case of server side config.